### ISSUE TITLE
System.IO.Tests.DirectoryInfo_Create.PathWithReservedDeviceNameAsExtendedPath failed in CI

### ISSUE THREAD

**Issue by hughbe** [dotnet/corefx#17763]
In #17717

Might be intermittent, so feel free to close if you want

```
Stacktrace

MESSAGE:
System.IO.IOException : The directory is not empty
+++++++++++++++++++
STACK TRACE:
at System.IO.Win32FileSystem.RemoveDirectoryHelper(String fullPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound) in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.FileSystem\src\System\IO\Win32FileSystem.cs:line 663 
at System.IO.Win32FileSystem.RemoveDirectory(String fullPath, Boolean recursive) in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.FileSystem\src\System\IO\Win32FileSystem.cs:line 509
at System.IO.Directory.Delete(String path, Boolean recursive) in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.FileSystem\src\System\IO\Directory.cs:line 581 
at TemporaryDirectory.Delete() in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.FileSystem\tests\PortedCommon\TemporaryDirectory.cs:line 20
 at TemporaryFileSystemItem`1.Dispose() in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.FileSystem\tests\PortedCommon\TemporaryFileSystemItem.cs:line 78 
at System.IO.Tests.Directory_CreateDirectory.PathWithReservedDeviceNameAsExtendedPath() in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.FileSystem\tests\Directory\CreateDirectory.cs:line 424
```

**Comment by karelz** [comment]
@hughbe FYI: We are trying to achieve clean test runs. Intermittent failures are waste of everyone's time - while the cost is small on the spot, it accumulates long-term (for both project maintainers and contributors).

We should always attempt to do something about intermittent failures (fix test/product, harden test against known flakiness of OS, add logging into test to know more next time, etc.).
There are 2 exceptions to the rule above:
1. Infrastructure failures (bad VMs, network/HW failure, etc.) as we can't harden the tests against all weird things. However, if a particular infrastructure failure is "common", we should go after it as well, detect it and flag it, to prevent future waste-of-time investments.
2. One-time test failures (as in fails once per year/ever) -- if it is not clear from the output what happened, it may be overkill to add logging, as it may be truly just some weird infrastructure issue. Larger investment may be waste of time. Waiting on second occurrence is a fine outcome, however, it should be the area owner who is making the decision (they know about other similar failures in the area and might connect the dots in failures which happen all across the board randomly).
    * Note: Functional test runs are NOT a good place to catch and track down reliability/stress issues. However, they may be good indicator of areas which may need some special reliability/stress-testing attention.

Of course all this goes under the typical cost/benefit/risk assessment. High cost (of root-causing the failure) with low benefit (i.e. test fails very rarely) with low risk (unlikely a real product issue) is a combo which can be pushed into future / given lower priority.
Issues which are either low cost (simple to add logging, or fix the problem, or just disable the test), or high benefit (the failure shows up in CI a lot), or high risk (it is likely real product issue) should get more attention and ideally should be addressed in 2.0 time-frame.

**Comment by danmosemsft** [comment]
@hughbe was this a failure in a core run or a netfx run? the latter would suggest a bug in netfx itself, which is less interesting/actionable.

I can't tell from your PR because I guess you rebased

**Comment by alexperovich** [comment]
@hughbe If this happens again can you reopen and link the Jenkins build. I can't find any runs that failed like this.
